qemu/tests/cpu_topology_test.py

```diff
@@ -0,0 +1,59 @@
+import random
+
+from avocado.utils import cpu
+
+from virttest import error_context
+from virttest import env_process
+
+from provider.cpu_utils import check_guest_cpu_topology
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Check guest gets correct vcpu num, cpu cores, processors, sockets, siblings
+
+    1) Boot guest with options: -smp n,cores=x,threads=y,sockets=z...
+    2) Check cpu topology
+
+    :param test: QEMU test object.
+    :param params: Dictionary with test parameters.
+    :param env: Dictionary with the test environment.
+    """
+
+    def check(p_name, exp, check_cmd):
+        """
+        Check the cpu property inside guest
+        :param p_name: Property name
+        :param exp: The expect value
+        :param check_cmd: The param to get check command
+        """
+        res = session.cmd_output(params[check_cmd]).strip()
+        if int(res) != int(exp):
+            test.fail('The vcpu %s number inside guest is %s,'
+                      ' while it is set to %s' % (p_name, res, exp))
+
+    vm_name = params['main_vm']
+    os_type = params['os_type']
+    vcpu_threads_list = [1, 2]
+    if params['machine_type'] == 'pseries':
+        vcpu_threads_list = [1, 2, 4, 8]
+    params['vcpu_cores'] = vcpu_cores = random.randint(1, 10)
+    host_cpu = cpu.online_count()
+    for vcpu_threads in vcpu_threads_list:
+        vcpu_sockets = min(max(host_cpu * 5 // (vcpu_cores * vcpu_threads), 1),
+                           random.randint(1, 10))
```

Comment:
Some hosts have a lot of CPU, so there is a risk:
```python
min(max(192 * 5 // (8 * 10), 1), random.randint(1, 10))
10
```
And then, `smp = vcpu_maxcpus = 8 * 10 * 10`
I think using fixed sockets (2) is okay for this test case.

IIUC, AMD CPUs do not support multiple threads except EYPC. @nanliu-r please help to confirm.

Reply:
> Some hosts have a lot of CPU, so there is a risk:
> 
> ```python
> min(max(192 * 5 // (8 * 10), 1), random.randint(1, 10))
> 10
> ```
> 
> And then, `smp = vcpu_maxcpus = 8 * 10 * 10`
> I think using fixed sockets (2) is okay for this test case.

@nanliu-r What do you think about this, to have a fixed sockets(2)?